data/random-teams.ts

```diff
@@ -559,22 +685,740 @@ export class RandomTeams {
+		case 'circlethrow':
+			return {cull: hasMove['stormthrow'] && !hasMove['rest']};
+		case 'drainpunch':
+			return {cull: hasMove['closecombat'] || (!hasType['Fighting'] && movePool.includes('swordsdance'))};
+		case 'dynamicpunch': case 'thunderouskick':
+			return {cull: hasMove['closecombat'] || hasMove['facade']};
+		case 'focusblast':
+			return {cull: movePool.includes('shellsmash') || hasRestTalk};
+		case 'hammerarm':
+			return {cull: hasMove['fakeout']};
+		case 'stormthrow':
+			return {cull: hasRestTalk};
+		case 'superpower':
+			return {
+				cull: hasMove['hydropump'] ||
+					(counter.Physical >= 4 && movePool.includes('uturn')) ||
+					(hasMove['substitute'] && !hasAbility['Contrary']),
+				isSetup: hasAbility['Contrary'],
+			};
+		case 'poisonjab':
+			return {cull: !hasType['Poison'] && counter.Status >= 2};
+		case 'earthquake':
+			const doublesCull = hasMove['earthpower'] || hasMove['highhorsepower'];
+			const movePoolCull = movePool.includes('bodypress') && movePool.includes('shellsmash');
+			const subToxicPossible = hasMove['substitute'] && movePool.includes('toxic');
+			return {cull: movePoolCull || (isDoubles && doublesCull) || subToxicPossible || hasMove['bonemerang']};
+		case 'scorchingsands':
+			return {cull: hasMove['willowisp'] || hasMove['earthpower'] || (hasMove['toxic'] && movePool.includes('earthpower'))};
+		case 'airslash':
+			return {
+				cull: movePool.includes('flamethrower') ||
+					(hasMove['hurricane'] && !counter.setupType) ||
+					hasRestTalk ||
+					(hasAbility['Simple'] && !!counter.recovery),
+			};
+		case 'bravebird':
+			return {cull: hasMove['dragondance']};
+		case 'hurricane':
+			return {cull: hasAbility['Tinted Lens'] && counter.setupType && !isDoubles};
+		case 'futuresight':
+			return {cull: hasMove['psyshock'] || hasMove['trick'] || movePool.includes('teleport')};
+		case 'photongeyser':
+			return {cull: hasMove['morningsun']};
+		case 'psychic':
+			return {cull: hasMove['psyshock'] && (counter.setupType || isDoubles)};
+		case 'psyshock':
+			const cullNoSetup = (hasAbility['Multiscale'] && !counter.setupType) || (hasAbility['Pixilate'] && counter.Special < 4);
+			return {cull: hasMove['psychic'] || (!counter.setupType && cullNoSetup) || (isDoubles && hasMove['psychic'])};
+		case 'bugbuzz':
+			return {cull: hasMove['uturn'] && !counter.setupType};
+		case 'leechlife':
+			return {cull: (isDoubles && hasMove['lunge']) || movePool.includes('firstimpression') || movePool.includes('spikes')};
+			break;
+		case 'stoneedge':
+			const gutsCullCondition = hasAbility['Guts'] && (!hasMove['dynamicpunch'] || hasMove['spikes']);
+			const rockSlidePlusStatusPossible = counter.Status && movePool.includes('rockslide');
+			return {cull: gutsCullCondition || rockSlidePlusStatusPossible || hasMove['rockblast'] || hasMove['rockslide']};
+		case 'poltergeist':
+			return {cull: hasMove['knockoff']};
```

Comment:
```suggestion
			// Special case for Dhelmise in Doubles, which doesn't want both
			return {cull: hasMove['knockoff']};
```
honestly just remove this and remove poltergeist from dhelm and that'd be fine too. simpler, not much impact. poltergeist sucks in doubles anyway